app/utilities/PersistenceManager.java

```diff
@@ -87,6 +88,23 @@ public static Token getToken(String token){
     }
 
     /**
+     * Persists restaurant into database.
+     *
+     * @param restaurant the Restaurant to be persisted
+     */
+    public static void saveRestaurant(Restaurant restaurant){
+        restaurant.getAddress().setRestaurantId(Restaurant.find.all().size() + 1);
```

Comment:
The thing I mentioned in previous comment was the source of that "complication", and of this one as well. Now it's all fixed.